src/Controller/CRUDController.php

```diff
@@ -294,12 +296,13 @@ public function deleteAction($id)
      *
      * @return Response|RedirectResponse
      */
-    public function editAction($id = null)
+    public function editAction($id = null) // NEXT_MAJOR: Remove the $id default value
     {
         $request = $this->getRequest();
         // the key used to lookup the template
         $templateKey = 'edit';
 
+        // NEXT_MAJOR: remove the next line and use the $id parameter of the function
```

Comment:
What about when editing a child? I guess `$this->admin->getIdParameter()` return `childId`, so this couldn't be replace by the `$id` argument of the function and I guess it wouldn't work.

Reply:
How can I test this situation ?

Reply:
[Having an admin -> child association](https://symfony.com/doc/master/bundles/SonataAdminBundle/reference/child_admin.html), the url for edit would be like: `/post/{id}/comment/{childId}/edit`

Reply:
I just test it, and indeed, it wouldn't work.